File: tasks/implementations/java/tasks/image_resizing.java
import java.nio.file.*;

class ImageResizing {
    public static void run(String size, String fixturesRoot) throws Exception {
        String[] tokens = Files.readString(Path.of(fixturesRoot, "generated", "image", size + ".ppm")).trim().split("\\s+");
        int width = Integer.parseInt(tokens[1]);
        int height = Integer.parseInt(tokens[2]);
        long checksum = 0;
        for (int y = 0; y < height / 2; y++) {
            for (int x = 0; x < width / 2; x++) {
                int idx = 4 + (((y * 2) * width + (x * 2)) * 3);
                checksum += Integer.parseInt(tokens[idx]) + Integer.parseInt(tokens[idx + 1]) + Integer.parseInt(tokens[idx + 2]);
            }
        }
        System.out.println(checksum);
    }
}

File: tasks/implementations/python/tasks/image_resizing.py
from pathlib import Path

def run(size: str, fixtures_root: str) -> None:
    tokens = Path(fixtures_root, "generated", "image", f"{size}.ppm").read_text(encoding="utf-8").split()
    width = int(tokens[1])
    height = int(tokens[2])
    data = list(map(int, tokens[4:]))
    checksum = 0
    for y in range(height // 2):
        for x in range(width // 2):
            idx = ((y * 2) * width + (x * 2)) * 3
            checksum += data[idx] + data[idx + 1] + data[idx + 2]
    print(checksum)
